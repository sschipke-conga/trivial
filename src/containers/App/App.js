import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


//components
import Form from '../Form/Form';
import Board from '../Board/Board'
import './App.scss';

class App extends Component {
  constructor() {
    super();
  }
  render() {
    const {teams,questions} = this.props
  return (
    <div className="app">
      {questions.length === 0 ? <Route exact path='/' render={(props) => <Form {...props} />} /> : <Route exact path='/' render={(props) => <Board {...props} />} />}
    </div>
  )}
}

export const mapStateToProps = (state) => ({
  teams: state.teams,
  questions: state.questions
})

export default connect(mapStateToProps)(App);