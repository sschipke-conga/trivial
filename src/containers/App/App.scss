*{
  box-sizing: border-box;
  margin:0;
  padding:0;
}

button:hover {
  cursor: pointer;
}
