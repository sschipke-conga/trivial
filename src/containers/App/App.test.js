import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import {App, mapStateToProps} from './App';

const mockQuestions = [{
  "category": "Sports",
  "type": "multiple",
  "difficulty": "medium",
  "question": "At which bridge does the annual Oxford and Cambridge boat race start?",
  "correct_answer": "Putney",
  "incorrect_answers": [
    "Hammersmith",
    "Vauxhall ",
    "Battersea"
  ]
},
{
  "category": "Entertainment: Video Games",
  "type": "multiple",
  "difficulty": "easy",
  "question": "Who is Sonic&#039;s sidekick?",
  "correct_answer": "Tails",
  "incorrect_answers": [
    "Shadow",
    "Amy",
    "Knuckles"
  ]
},
{
  "category": "Politics",
  "type": "boolean",
  "difficulty": "easy",
  "question": "There was a satirical candidate named &quot;Deez Nuts&quot; running in the 2016 US presidential elections.",
  "correct_answer": "True",
  "incorrect_answers": [
    "False"
  ]
}]
const mockCurrentQuestion = {
  "category": "Sports",
  "type": "multiple",
  "difficulty": "medium",
  "question": "At which bridge does the annual Oxford and Cambridge boat race start?",
  "correct_answer": "Putney",
  "incorrect_answers": [
    "Hammersmith",
    "Vauxhall ",
    "Battersea"
  ]
}

const mockTeams = [{
  name: 'Lez Quizzerable',
  score: 0
}, {
  name: "Let's Get Quizzacle",
  score: 0
}]

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      questions={mockQuestions}
      haveWinner={false}
      haveQuestions={true}
      teams={mockTeams}
      />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('mapStateToProps gives all the questions, teams, and turnCount in state', () => {
    const mockState = {
      questions: mockQuestions,
      currentQuestion: mockCurrentQuestion,
      teams: mockTeams,
      haveQuestions: true,
      haveWinner: false
    };
    const expected = {
      questions: mockState.questions,
      teams: mockState.teams,
      currentQuestion: mockState.currentQuestion,
      haveQuestions: true,
      haveWinner: false
    };
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })
})

describe('App and EndGame', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      haveWinner={true}
      teams={mockTeams}
      questions={[]}
      haveQuestions={true}
    />)
  })
  it('should render the EndGame component if there is a winner', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('App and Form', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      haveWinner={false}
      teams={[]}
      questions={[]}
      haveQuestions={false}
    />)
  })
  it('should render the Form component if there are no teams and no questions', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('App and loading image', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App
      haveWinner={false}
      teams={['a', 'b']}
      questions={[]}
      haveQuestions={false}
    />)
  })
  it('should render the loading image if there are teams but no questions', () => {
    expect(wrapper).toMatchSnapshot()
  })
})