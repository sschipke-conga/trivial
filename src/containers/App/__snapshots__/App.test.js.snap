// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App and EndGame should render the EndGame component if there is a winner 1`] = `
<div
  className="app"
>
  <Connect(EndGame) />
</div>
`;

exports[`App and Form should render the Form component if there are no teams and no questions 1`] = `
<div
  className="app"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</div>
`;

exports[`App should match the snapshot 1`] = `
<div
  className="app"
>
  <Route
    path="/"
    render={[Function]}
  />
</div>
`;
